grid_generation/Gen_CICE_grid.py

```diff
@@ -0,0 +1,167 @@
+"""
+Script: Gen_CICE_grid.py
+
+Description: 
+This script generates a CICE grid from the MOM super grid provided in the input NetCDF file.
+
+Contact: 
+Name: Ezhilsabareesh Kannadasan
+
+Usage:
+python generate_cice_grid.py <input_superGridPath> <output_file>
+- input_superGridPath: Path to the MOM super grid NetCDF file.
+- output_file: Path to the output CICE grid NetCDF file.
+"""
+import numpy as np
+import xarray as xr
+from netCDF4 import Dataset
+import sys
+import os
+import subprocess
+from datetime import datetime
+
+
+def is_git_repo():
+    """
+    Return True/False depending on whether or not the current directory is a git repo.
+    """
+
+    return subprocess.call(
+        ['git', '-C', '.', 'status'],
+        stderr=subprocess.STDOUT,
+        stdout = open(os.devnull, 'w')
+    ) == 0
+
+def git_info():
+    """
+    Return the git repo origin url, relative path to this file, and latest commit hash.
+    """
+
+    url = subprocess.check_output(
+        ["git", "remote", "get-url", "origin"]
+    ).decode('ascii').strip()
+    top_level_dir = subprocess.check_output(
+        ['git', 'rev-parse', '--show-toplevel']
+    ).decode('ascii').strip()
+    rel_path = os.path.relpath(__file__, top_level_dir)
+    hash = subprocess.check_output(
+        ['git', 'rev-parse', 'HEAD']
+    ).decode('ascii').strip()
+
+    return url, rel_path, hash
+
+def generate_cice_grid(in_superGridPath, output_file):
+    # Read input files
+    in_superGridFile = xr.open_dataset(in_superGridPath)
+
+    # Constants
+    ULAT = np.deg2rad(in_superGridFile['y'][2::2, 2::2])
+    ULON = np.deg2rad(in_superGridFile['x'][2::2, 2::2])
+    TLAT = np.deg2rad(in_superGridFile['y'][1::2, 1::2])
+    TLON = np.deg2rad(in_superGridFile['x'][1::2, 1::2])
+
+    HTN = in_superGridFile['dx'] * 100.0  # convert to cm
+    HTN = HTN[1::2, ::2] + HTN[1::2, 1::2]
```

Comment:
![Screenshot 2024-03-07 at 4 15 14 pm](https://github.com/COSIMA/om3-scripts/assets/79179784/f2345540-42e9-4e7b-a57d-71a59d90b956)


In the mom_grid, dx has units ('nyp','nx'). These correspond to the edges in y and the middle in x. So, the y coordinate (i.e. rows) should be 2::2